ui/src/cloud/actions/demodata.ts

```diff
@@ -60,85 +64,72 @@ export const getDemoDataBuckets = () => async (
-  dispatch,
-  getState: GetState
-) => {
+export const getDemoDataBucketMembership = ({
+  name: bucketName,
+  id: bucketID,
+}) => async (dispatch, getState: GetState) => {
   const state = getState()
+
   const {
     me: {id: userID},
   } = state
+
   const {id: orgID} = getOrg(state)
 
   try {
-    await getDemoDataBucketMembershipAJAX(bucket.id, userID)
+    await getDemoDataBucketMembershipAJAX(bucketID, userID)
 
-    const template = await DemoDataTemplates[bucket.name]
-    if (template) {
-      await createDashboardFromTemplate(template, orgID)
-    } else {
+    const normalizedBucket = await getNormalizedDemoDataBucket(bucketID)
+    dispatch(addBucket(normalizedBucket))
+
+    const template = await DemoDataTemplates[bucketName]
+    if (!template) {
       throw new Error(
-        `Could not find template for demodata bucket ${bucket.name}`
+        `Could not find dashboard template for demodata bucket ${bucketName}`
       )
     }
 
-    const resp = await getBucket({bucketID: bucket.id})
-
-    if (resp.status !== 200) {
-      throw new Error('Request for demo data bucket membership did not succeed')
-    }
+    await createDashboardFromTemplate(template, orgID)
+    const updatedState = getState()
 
-    const newBucket = {
-      ...resp.data,
-      type: 'demodata' as 'demodata',
-      labels: [],
-    } as DemoBucket
+    const allDashboards = getAll<Dashboard>(
+      updatedState,
+      ResourceType.Dashboards
+    )
 
-    const normalizedBucket = normalize<Bucket, BucketEntities, string>(
-      newBucket,
-      bucketSchema
+    const createdDashboard = allDashboards.find(
+      d => d.name === DemoDataDashboards[bucketName]
     )
 
-    dispatch(addBucket(normalizedBucket))
+    if (!createdDashboard) {
+      throw new Error(
+        `Could not create dashboard for demodata bucket ${bucketName}`
+      )
+    }
+
+    const url = `/orgs/${orgID}/dashboards/${createdDashboard.id}`
 
-    // TODO: notify success and error appropriately
+    dispatch(notify(demoDataSucceeded(bucketName, url)))
   } catch (error) {
-    console.error(error)
-  }
-}
-export const deleteDemoDataDashboard = (dashboardName: string) => async (
-  dispatch,
-  getState: GetState
-) => {
-  try {
-    await dispatch(getDashboards())
+    dispatch(notify(demoDataAddBucketFailed(error)))
 
-    const updatedState = getState()
+    reportError(error, {
+      name: 'getDemoDataBucketMembership function',
+    })
 
-    const ddDashboard = getAll(updatedState, ResourceType.Dashboards).find(
-      d => {
-        d.name === dashboardName
-      }
-    ) as Dashboard
-
-    if (ddDashboard) {
-      const deleteResp = await deleteDashboard({
-        dashboardID: ddDashboard.id,
-      })
-      if (deleteResp.status !== 204) {
-        throw new Error(deleteResp.data.message)
-      }
-    }
-  } catch (error) {
-    throw new Error(error)
+    console.error(error)
```

Comment:
same suggestion here for removing this - not sure if this is valuable since we're reporting the error. but it might be and i may be a dunce